test/unit/io/sequence_file/sequence_file_output_test.cpp

```diff
@@ -426,6 +454,28 @@ TEST(columns, assign_tuple_of_columns)
     assign_impl(std::tie(seqs, ids));
 }
 
+TEST(columns, writing_seq_qual)
+{
+    sequence_file_output fout{std::ostringstream{}, sequence_file_format_fasta{}, fields<field::ID, field::SEQ_QUAL>()};
+    fout.options.fasta_letters_per_line = 0;
+
+    std::vector<std::vector<qualified<dna5, phred42>>> seq_quals{3};
+    for (size_t i = 0; i < 3; ++i)
+    {
+        seq_quals[i].resize(quals[i].size());
+        for (size_t j = 0; j < quals[i].size(); ++j)
+        {
+            seq_quals[i][j] = seqs[i][j];
+            seq_quals[i][j] = quals[i][j];
+        }
```

Comment:
here you need the vector of vectors, but you might also be able to use std::copy